File: sdk/python/src/buildkite_sdk/sdk.py
from buildkite_sdk.schema import CommandStep
from buildkite_sdk.environment import Environment
import json
import yaml

class Pipeline:
    def __init__(self):
        self.steps = []

    def add_command_step(self, props: CommandStep):
        self.steps.append(props)

    def to_json(self):
        return json.dumps(self.__dict__)

    def to_yaml(self):
        return yaml.dump(self.__dict__)

File: sdk/typescript/src/sdk.ts
import * as yaml from "yaml";
import * as schema from "./schema";
import * as env from "./environment";
export * as environmentVariables from "./environment";

type PipelineStep =
    | schema.CommandStep
    | schema.WaitStep
    | schema.InputStep
    | schema.TriggerStep
    | schema.BlockStep
    | schema.GroupStepClass;

// export class Environment {
//     static get(key: env.Environment) {
//         env.Environment.BUILDKITE_AGENT_DISCONNECT_AFTER_IDLE_TIMEOUT;
//         return env.Environment[key];
//     }
// }

export class Pipeline {
    private steps: PipelineStep[] = [];

    constructor() {
        this.steps = [];
    }

    /**
     * Add a step to the pipeline.
     * @param step
     * @returns
     */
    addStep(step: PipelineStep) {
        this.steps.push(step);
        return this;
    }

    toJSON() {
        return JSON.stringify(
            {
                steps: this.steps,
            },
            null,
            4
        );
    }

    toYAML() {
        return yaml.stringify({
            steps: this.steps,
        });
    }
}
